### PR TITLE
Introducing percentage type

### LINKED ISSUE #2073: Using other rich defined type instead of primitive int.
https://github.com/pipe-cd/pipe/blob/06a9629150c5bf95fff11da9f3ecef4300098de8/pkg/config/deployment_ecs.go#L61-L64

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

Scale configuration from ECS canary rollout stage option is now accepted not only number but also string number with suffix `%`

```yaml
  pipeline:
    stages:
      - name: ECS_CANARY_ROLLOUT
        with:
          scale: 30%
```

**Which issue(s) this PR fixes**:

Fixes #2073 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/config/percentage.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		if strings.HasSuffix(raw, "%") {
+			percentage.HasSuffix = true
+			raw = strings.TrimSuffix(raw, "%")
+		}
+		value, err := strconv.ParseInt(raw, 10, 64)
+		if err != nil {
+			return fmt.Errorf("invalid percentage: %v", err)
+		}
+		percentage.Number = int(value)
+		*p = percentage
```
Comment (nghialv) [inline pkg/config/percentage.go:70]: nit: We can use only this part and remove lines from 46 to 57. (And replace `raw` with `string(b)`).
It is up to you.

Comment (khanhtc1202) [reply pkg/config/percentage.go]: You're right, it looks far better now, thx 🙏 
Addressed by https://github.com/pipe-cd/pipe/pull/2088/commits/8e58eabcbf469aaee70fe78155acc68cd30894af

File: pkg/config/percentage.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+func (p Percentage) Int() int {
+	return p.Number
+}
+
+func (p Percentage) MarshalJSON() ([]byte, error) {
+	return json.Marshal(p.String())
+}
+
+func (p *Percentage) UnmarshalJSON(b []byte) error {
+	raw := strings.Trim(string(b), "\"")
```
Comment (nghialv) [inline pkg/config/percentage.go:46]: nit nit: 
```
`"`
```

Comment (khanhtc1202) [reply pkg/config/percentage.go]: :ok

File: pkg/config/percentage.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	percentage := Percentage{
+		HasSuffix: false,
+	}
+	if strings.HasSuffix(raw, "%") {
+		percentage.HasSuffix = true
+		raw = strings.TrimSuffix(raw, "%")
+	}
+	value, err := strconv.ParseInt(raw, 10, 64)
+	if err != nil {
+		return fmt.Errorf("invalid percentage: %v", err)
```
Comment (nakabonne) [inline pkg/config/percentage.go:56]: nit
```suggestion
		return fmt.Errorf("invalid percentage: %w", err)
```

Comment (khanhtc1202) [reply pkg/config/percentage.go]: Thx, adopted 🙆‍♀️

**Comment by nakabonne** [comment]
Thank you!
/lgtm

**Comment by nghialv** [comment]
Nice.
/approve

**Comment by khanhtc1202** [comment]
@nakabonne @nghialv PTAL 🙏 
Just fix the broken test after a change from `%v` to `%w` on returning error format.

**Pull request merged**